src/diffusers/models/unet_2d_blocks.py

```diff
@@ -1485,7 +1516,7 @@ def forward(
             hidden_states = attn(
                 hidden_states,
                 encoder_hidden_states=encoder_hidden_states,
-                attention_mask=attention_mask,
+                attention_mask=attention_mask if encoder_hidden_states is None else encoder_attention_mask,
```

Comment:
As is, this is actually unfortunately not backwards compatible. This block already takes a single attention mask through `attention_mask`. As this block just performs a single either self or cross attention, it doesn't make sense for this block to take two possible masks. Concretely, this will cause the attention mask in the unclip pipeline to not be used.

Reply:
Great catch!

Reply:
This is a great observation! But tests passed, is that because of the unclip pipeline being so fickle? Would it make sense to revisit those tests after @sayakpaul's recent efforts to increase determinism in tests?

Reply:
Good point @pcuenca! I would have to take a look :grin: